base/acme/src/main/java/org/dogtagpki/acme/server/ACMEEngineConfigFileSource.java

```diff
@@ -96,41 +98,91 @@ public void run() {
 
         final Path path = FileSystems.getDefault().getPath(filename);
 
-        try (WatchService watchService = FileSystems.getDefault().newWatchService()) {
-            // You can only watch a directory, not a file.  So we must
-            // watch the parent dir and check the event is for the file.
-            path.getParent().register(watchService, StandardWatchEventKinds.ENTRY_MODIFY);
-            while (true) {
-                WatchKey wk = watchService.take();
-                for (WatchEvent<?> event : wk.pollEvents()) {
-                    if (event.kind() == StandardWatchEventKinds.ENTRY_MODIFY) {
-                        if (((Path) event.context()).endsWith(path.getFileName())) {
-                            loadFile();
+        // how long to sleep before retrying
+        int delay_ms = 0;
+
+        while (monitor) {
+
+            if (delay_ms > 0) {
+                try {
+                    logger.info("ACMEEngineConfigSource: sleeping for " + delay_ms + "ms");
+                    Thread.sleep(delay_ms);
+                } catch (InterruptedException e) {
+                    // no big deal
+                }
+
+                // We looped due to an exception.  It is possible that the file
+                // changed, and that we were unable to read it, but after the sleep
+                // it might be OK.  So let's try and read it, but ignore failures.
+                try {
+                    loadFile();
+                } catch (Throwable e) {
+                    // ignore
```

Comment:
Could we log a warning with the exception message (instead of stack trace) so at least we know when/why the file couldn't be loaded? For example:
```
logger.warn("ACMEEngineConfigSource: Unable to load " + filename + ": " + e.getMessage);
```

Reply:
Done